File: src/main/java/com/bdev/hogwarts_api/service/student/StudentService.java
package com.bdev.hogwarts_api.service.student;

import com.bdev.hogwarts_api.data.dto.student.Student;

import java.util.List;
import java.util.Optional;

public interface StudentService {
    boolean exists(long studentId);
    List<Student> getAllStudents();
    Optional<Student> getStudentById(long studentId);
    long createStudent(Student student);
    void editStudent(Student student);
    void deleteStudent(long studentId);
}

File: src/main/java/com/bdev/hogwarts_api/service/student/StudentService.kt
package com.bdev.hogwarts_api.service.student

import com.bdev.hogwarts_api.data.dto.student.Student

interface StudentService {
    fun getAllStudents(): List<Student>
    fun exists(studentId: Long): Boolean
    fun getStudentById(studentId: Long): Student?
    fun createStudent(student: Student): Long
    fun editStudent(student: Student)
    fun deleteStudent(studentId: Long)
}
